src/lamp_py/performance_manager/alerts.py

```diff
@@ -125,67 +111,64 @@ def append_new_records(self, alerts: pandas.DataFrame) -> None:
         """
         append alerts to the end of the local parquet file using batches to keep memory usage lower
         """
-        self.new_data = True
+        process_logger = ProcessLogger(
+            process_name="append_new_alerts_records",
+        )
+        process_logger.log_start()
+
         alerts = alerts.reset_index(drop=True)
         alerts_table = pyarrow.Table.from_pandas(
             alerts, schema=self.parquet_schema
         )
 
-        # if there is no local file, write the alerts to the local file and exit
-        if not self.local_exists:
-            with pq.ParquetWriter(
-                self.local_path, schema=self.parquet_schema
-            ) as writer:
-                writer.write_table(alerts_table)
-
-            self.local_exists = True
+        if alerts_table.num_rows == 0:
+            process_logger.log_complete()
             return
 
+        new_dataset = pd.dataset(alerts_table)
+
+        joined_ds = new_dataset
+        if os.path.exists(self.local_path):
+            joined_ds = pd.dataset([pd.dataset(self.local_path), new_dataset])
+
+        process_logger.add_metadata(
+            new_records=new_dataset.count_rows(),
+            total_records=joined_ds.count_rows(),
+        )
+
         # write new alerts to a new file and then join them with the local file in batches
         new_path = os.path.join("/tmp", "new_alerts.parquet")
-        joined_path = os.path.join("/tmp", "joined_alerts.parquet")
+        row_group_count = 0
 
-        with pq.ParquetWriter(new_path, schema=self.parquet_schema) as writer:
-            writer.write_table(alerts_table)
-
-        joined_batches = pd.dataset(
-            [
-                pd.dataset(self.local_path),
-                pd.dataset(new_path),
-            ]
-        ).to_batches()
-
-        with pq.ParquetWriter(
-            joined_path, schema=self.parquet_schema
-        ) as writer:
-            for batch in joined_batches:
-                writer.write_batch(batch)
-
-        # the joined file becomes the local file and we can delete the new and
-        # joined files
-        os.replace(joined_path, self.local_path)
-        os.remove(new_path)
-
-    def download_data(self) -> bool:
-        """
-        download the s3 parquet file to a temp location if the versions match.
-        return True if the file was downloaded successfully, False otherwise.
-        """
-        try:
-            lamp_version = object_metadata(self.s3_path).get(
-                self.version_key, ""
-            )
+        now = datetime.now(tz=BOSTON_TZ)
+        start = datetime(2028, 1, 1, tzinfo=BOSTON_TZ)
```

Comment:
I would recommend against using a tz here, other than `UTC`, since the timestamp field we're partitioning on is a UTC value, would probably make debugging issues in partition more difficult. 

Also shouldn't need to hard-code a start datetime here, could use `pyarrow.compute.min` function on `partition_key` column of the dataset to pull this value dynamically. 

Using `now` for the final partition could also be an issue, in my testing there are a huge number of alerts that are projected to start past the current date, So maybe also pulling the max value from `partition_key` column and doing the partitions dynamically (maybe every 3 months) would be a better approach.